### ISSUE TITLE
possibly incorrect build_prefix?

### ISSUE THREAD

**Issue by kalefranz** [conda/conda-build#2583]
Building conda with 3.1.4 and master, I get something like this (line numbers from current master).  Using conda-build 3.1.2, trying to build conda 4.3.31 still builds fine.

```
Installed /usr/local/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/conda-bld/conda_1513369440994/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.6/site-packages/conda-4.3.31-py3.6.egg
Processing dependencies for conda==4.3.31
Finished processing dependencies for conda==4.3.31
Traceback (most recent call last):
  File "/conda/bin/conda-build", line 6, in <module>
    sys.exit(main())
  File "/Users/kfranz/continuum/conda-build/conda_build/cli/main_build.py", line 393, in main
    execute(sys.argv[1:])
  File "/Users/kfranz/continuum/conda-build/conda_build/cli/main_build.py", line 384, in execute
    verify=args.verify)
  File "/Users/kfranz/continuum/conda-build/conda_build/api.py", line 186, in build
    need_source_download=need_source_download, config=config, variants=variants)
  File "/Users/kfranz/continuum/conda-build/conda_build/build.py", line 1847, in build_tree
    notest=notest,
  File "/Users/kfranz/continuum/conda-build/conda_build/build.py", line 1173, in build
    utils.remove_pycache_from_scripts(m.config.build_prefix)
  File "/Users/kfranz/continuum/conda-build/conda_build/utils.py", line 1225, in remove_pycache_from_scripts
    for entry in os.listdir(scripts_path):
OSError: [Errno 2] No such file or directory: '/usr/local/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/conda-bld/conda_1513369440994/_build_env/bin'
```

**Comment by kalefranz** [comment]
This is using the recipe at

https://github.com/AnacondaRecipes/conda-feedstock/blob/690f7aff4542749486ab971a3848d184de2ac271/recipe/meta.yaml

**Comment by msarahan** [comment]
fixed by #2584
